lib/ansible/module_utils/tacp_ansible/tacp_utils.py

```
@@ -6,148 +6,205 @@
+    return wrapper
 
 
-class ApplicationResource(ApiResource):
+class Resource(object):
+    resource_class = None
 
-    model = tacp.ApplicationsApi
+    def __init__(self, client):
+        assert self.resource_class is not None
+        self.api = self.resource_class(client)
+        self._api_client = client
 
-    def _get_all(self, *args, **kwargs):
-        if kwargs:
-            return self._api.get_applications_using_get(filters=";".join(kwargs))
-        else:
-            return self._api.get_applications_using_get()
+    def get_filters_query_string(self, **kws):
+        """
+        Returns a string used as filters query string
+
+        To use operators provide the value of a kwarg as two item tuple:
+            (operator, actual_value)
+
+        If the value of a kwarg is not a two item tuple then the default `==`
+        operator is used.
+
+        Operators:
+            - Equal to: ==
+            - Not equal to: !=
+            - Less than: =lt= or <
+            - Less than or equal to: =le= or <=
+            - Greater than operator: =gt= or >
+            - Greater than or equal to: =ge= or >=
+            - In: =in=
+            - Not in: =out=
+
+        The following:
+            name='some name'
+
+        Is the same as:
+            name=('==', 'some name')
+        """
+        if not kws:
+            return ''
 
-    def _get_by_uuid(self, uuid):
-        return self._api.get_application_using_get(uuid)
+        allowed = ['==', '!=', '=lt=', '<', '=le=', '<=', '=gt=', '>', '=ge=',
+                   '>=', '=in=', '=out=']
 
-    def _create(self, body):
-        return self._api.create_application_from_template_using_post(body)
+        filters = []
 
-    def _delete(self, uuid):
-        return self._api.delete_application_using_delete(uuid)
+        for k, v in kws.items():
+            if isinstance(v, (list, tuple)):
+                op, value = v
+                if op not in allowed:
+                    continue
```

If an op is not allowed, it seems that you should throw an exception instead of skip and continue.

---

Initially I was thinking to just log some error message, but raising an exception would probably be better. Done.